File: springboot2.x-learning/springboot-data-jpa-multi/src/main/java/com/rtxtitanv/repository/primary/PrimaryUserRepository.java
package com.rtxtitanv.repository.primary;

import com.rtxtitanv.model.primary.PrimaryUser;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

/**
 * @author rtxtitanv
 * @version 1.0.0
 * @name com.rtxtitanv.repository.primary.PrimaryUserRepository
 * @description PrimaryUserRepository接口用于操作主库用户表
 * @date 2020/1/7 19:34
 */
public interface PrimaryUserRepository extends JpaRepository<PrimaryUser, Long>, JpaSpecificationExecutor<PrimaryUser> {
}

File: springboot2.x-learning/springboot-data-jpa-multi-kotlin/src/main/kotlin/com/rtxtitanv/repository/primary/PrimaryUserRepository.kt
package com.rtxtitanv.repository.primary

import com.rtxtitanv.model.primary.PrimaryUser
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor

/**
 * @name com.rtxtitanv.repository.primary.PrimaryUserRepository
 * @description PrimaryUserRepository接口用于操作主库用户表
 * @author rtxtitanv
 * @date 2020/2/5 19:01
 * @version v1.0.0
 */
interface PrimaryUserRepository : JpaRepository<PrimaryUser, Long>, JpaSpecificationExecutor<PrimaryUser>